storm/ia32/system_calls.rb

```diff
@@ -0,0 +1,170 @@
+#include <storm/ia32/wrapper.h>
+")
+
+    system_calls.each do |system_call, num_parameters|
+      file.puts %Q[
+void wrapper_#{system_call}(void)
+{
+  asm ("pushal\\n\"]
+
+      file.puts %Q[
+                // Push all arguments. This approach pretty smart; it utilizes the fact that the stack grows downwards
+                // so the "next parameter to push" is always in the same memory location. :)
+      ]
+
+      for parameter in 0..num_parameters - 1 do
+        file.puts "                \"pushl  32 + 4 + #{num_parameters} * 4(%esp)\\n\"\n"
+      end
+
+      file.puts %Q[\
+                "call  system_call_#{system_call}\\n"
+
+                "addl  \$4 * #{num_parameters}, %esp\\n"
+
+               // Simulate a popa, without overwriting EAX (since it contains the return value from the system call).
+               "popl   %edi\\n"
+               "popl   %esi\\n"
+               "popl   %ebp\\n"
+
+               // ESP can't be popped for obvious reasons.
+               "addl   \$4, %esp\\n"
+               "popl   %ebx\\n"
+               "popl   %edx\\n"
+               "popl   %ecx\\n"
+
+               // Adjust the stack for the fact that EAX isn't being popped.
+               "addl   \$4, %esp\\n"
+               "lret   \$4 * #{num_parameters}\\n");
+}
+]
+    end
+  end
+end
+
+def create_include_storm_system_calls_h(system_calls)
+  File.open('../include/storm/system_calls.h', 'wb') do |file|
+    file.puts(
+"// Generated automatically by system_calls.rb. Do not modify!
+
+#pragma once
+
+#define SYSTEM_CALLS #{system_calls.keys.count}
+
+")
+    
+    file.puts "enum\n{\n  SYSTEM_CALL_#{system_calls.keys.first.upcase} = #{$gdt_start},\n"
+
+    system_calls.keys[1...system_calls.keys.count].each do |system_call|
+      file.puts "  SYSTEM_CALL_#{system_call.upcase},\n"
+    end
+
+    file.puts "};\n"
+  end
+end
+
+def create_system_calls_auto_c(system_calls)
+  File.open('system_calls-auto.c', 'wb') do |file|
+    file.puts "// Generated automatically by system_calls.rb. Do not modify!
+
+#include <storm/ia32/system_calls.h>
+#include <storm/ia32/wrapper.h>
+
+const system_call_type system_call[] =
+{
+"
+
+    system_calls.each do |system_call, num_parameters|
+      file.puts "  { SYSTEM_CALL_#{system_call.upcase}, wrapper_#{system_call}, #{num_parameters} },\n"
+    end
+
+    file.puts "};\n"
+  end
+end
+
+def create_include_storm_ia32_wrapper_h(system_calls)
+  File.open('../include/storm/ia32/wrapper.h', 'wb') do |file|
+    file.puts "// Generated automatically by system_calls.rb. Do not modify!
+
+#pragma once
+
+"
+
+    system_calls.keys.each { |system_call| file.puts "void wrapper_#{system_call}(void);\n" }
+  end
+end
+
+$0.sub! 'system_calls.rb', ''
+Dir.chdir($0) or fail "Couldn't change directory: $!"
+
+create_wrapper_c system_calls
+create_include_storm_system_calls_h system_calls
+create_system_calls_auto_c system_calls
+create_include_storm_ia32_wrapper_h system_calls
```

Comment:
This could arguably be done even cleaner and nicer using [ERB](http://www.stuartellis.eu/articles/erb/) syntax. I would be very glad to incorporate changes in that direction; PR:s are welcome. However, for now I just did a pretty much 1-to-1 conversion (but changing to more Rubish syntax (`.each` etc), and using a hash instead of an plan array etc). But changing to ERB would be even nicer. I consider this "good enough" for now though.